micro_dl/cli/metrics_script.py

```diff
@@ -1,263 +1,138 @@
+        "ssim",
+        "corr",
+        "r2",
+        "mse",
+        "mae",
+        "acc",
+        "dice",
+        "IoU",
+        "VI",
+        "POD",
+    ]
+    d_pod = [
+        "OD_true_positives",
+        "OD_false_positives",
+        "OD_false_negatives",
+        "OD_precision",
+        "OD_recall",
+        "OD_f1_score",
+    ]
+
+    metric_map = {
+        "mae_metric": metrics.mae_metric,
+        "mse_metric": metrics.mse_metric,
+        "r2_metric": metrics.r2_metric,
+        "corr_metric": metrics.corr_metric,
+        "ssim_metric": metrics.ssim_metric,
+        "acc_metric": metrics.accuracy_metric,
+        "dice_metric": metrics.dice_metric,
+        "IoU_metric": metrics.IOU_metric,
+        "VI_metric": metrics.VOI_metric,
+        "POD_metric": metrics.POD_metric,
     }
-    df_mapping = {
-        'xy': pd.DataFrame(),
-        'xz': pd.DataFrame(),
-        'yz': pd.DataFrame(),
-        'xyz': pd.DataFrame(),
-    }
-
-    # Iterate over all indices for test data
-    for time_idx in metadata_ids['time_idx']:
-        for pos_idx in metadata_ids['pos_idx']:
-            target_stack = []
-            pred_stack = []
-            for slice_idx in metadata_ids['slice_idx']:
-                im_idx = aux_utils.get_meta_idx(
-                    frames_metadata=frames_meta,
-                    time_idx=time_idx,
-                    channel_idx=target_channel,
-                    slice_idx=slice_idx,
-                    pos_idx=pos_idx,
-                )
-                target_fname = os.path.join(
-                    image_dir,
-                    frames_meta.loc[im_idx, 'file_name'],
-                )
-                im_target = image_utils.read_image(target_fname)
-                im_target = im_target.astype(np.float32)
 
-                pred_fname = aux_utils.get_im_name(
-                    time_idx=time_idx,
-                    channel_idx=pred_channel,
-                    slice_idx=slice_idx,
-                    pos_idx=pos_idx,
-                    ext=ext,
+    pred_plate = ngff.open_ome_zarr(store_path=pred_dir, mode="r+")
+    chan_names = pred_plate.channel_names
+
+    metric_chan_mask = metric_channel + "_cp_mask"
+    path_split_head_tail = os.path.split(pred_dir)
+    target_zarr_dir = path_split_head_tail[0]
+    ground_truth_dir = os.path.join(target_zarr_dir, ground_truth_subdir)
+
+    df_metrics = pd.DataFrame(columns=available_metrics[:-1] + d_pod[:-1])
+
+    for i, (position, pos_data) in enumerate(pred_plate.positions()):
+        pos = int(position.split("/")[2])
+
+        if pos in PosList:
+            raw_data = pos_data.data
+            target_data = raw_data[0, chan_names.index(metric_chan_mask), ...]
+
+            gt_mask_save_name = (
+                "^" + metric_channel + "_p" + str(format(pos, "03d")) + "_z"
+            )
+            z_slice_no = int(
+                glob.glob(ground_truth_dir + "/" + gt_mask_save_name + "*_cp_mask.png")
```

Comment:
This will not work on Windows. `os.path.join` is more robust to file system variations.